crud/common/sharding/init.lua

```diff
@@ -185,4 +185,50 @@ function sharding.wrap_select_method(method, space_name, ...)
     return res, err
 end
 
+--- Split tuples by replicaset for specified space
+--
+-- @function split_tuples_by_replicaset
+--
+-- @param table tuples
+--  Tuples to split
+--
+-- @param table space
+--  Specified space
+--
+-- @return[1] batches
+--  Map where key is a replicaset and value
+--  is table of tuples related to this replicaset
+function sharding.split_tuples_by_replicaset(tuples, space)
+    dev_checks('table', 'table')
+
+    local batches = {}
+
+    local sharding_data
+    local err
+    for _, tuple in ipairs(tuples) do
+        sharding_data, err = sharding.tuple_set_and_return_bucket_id(tuple, space)
+        if err ~= nil then
+            return nil, BucketIDError:new("Failed to get bucket ID: %s", err)
+        end
+
+        local replicaset, err = vshard.router.route(sharding_data.bucket_id)
+        if replicaset == nil then
+            return nil, GetReplicasetsError:new(
+                    "Failed to get replicaset for bucket_id %s: %s",
+                    sharding_data.bucket_id, err.err)
+        end
+
+        local tuples_by_replicaset = batches[replicaset] or {}
+        table.insert(tuples_by_replicaset, tuple)
+        batches[replicaset] = tuples_by_replicaset
+    end
+
+    return {
+        batches = batches,
+        sharding_func_hash = sharding_data.sharding_func_hash,
```

Comment:
This approach won't work if several records in a batch have `bucket_id` set and some are missing. It should be something like 

```
local sharding_func_hash = nil

for ...
    if sharding_data.sharding_func_hash ~= nil then
        sharding_func_hash = sharding_data.sharding_func_hash
    end 
end
```

Be careful with `skip_sharding_hash_check` -- if at least one record do not allow skipping, it must be `false` (or `nil`)

Reply:
Thanks! Fixed I hope.